### PR TITLE
Add support for transcoding recordings

### PR DESCRIPTION
**Pull request by dontcallmedom**
Between MP3 & WebM
Follow up from #21

### CODE REVIEW

File: public/recorder.js [hunk 1]
```diff
@@ -117,7 +117,12 @@ recordBtn.addEventListener("click", async () => {
   ref.pause();
   ref.currentTime = 0;
   stream = await navigator.mediaDevices.getUserMedia({audio: true});
-  recorder = new MediaRecorder(stream, {mimeType: "audio/webm"});
+  let mimeType = "audio/webm";
+  if (!MediaRecorder.isTypeSupported("audio/webm")) {
+    mimeType = "audio/mpeg";
```
Comment (dontcallmedom) [inline public/recorder.js:122]: That's not a credible alternative a priori given https://github.com/dontcallmedom/canon-singing/pull/21#issuecomment-696580283

Comment (dontcallmedom) [reply public/recorder.js:122]: although it would be with the polyfill? https://github.com/ai/audio-recorder-polyfill#mp3

File: index.js [hunk 2]
```diff
@@ -28,12 +29,16 @@ app.post('/upload', function(req, res) {
...
   let coverFile = req.files.cover;
+  let format = req.body.format || "audio/webm";
+  let extension = format === "audio/webm" ? ".webm" : ".mp3";
+  let transcodedExtension = extension === ".webm" ? ".mp3" : ".webm";
   // generate random name
   let shortname = nanoid();
-  let name =  shortname + ".mp3";
 
+  let audioPath = path.join(__dirname, '_submissions/audio/' + shortname + extension);
+  let transcodedAutioPath = path.join(__dirname, '_submissions/audio/' + shortname + transcodedExtension);
```
Comment (tidoust) [inline index.js]:
```suggestion
  let transcodedAudioPath = path.join(__dirname, '_submissions/audio/' + shortname + transcodedExtension);
```

File: index.js [hunk 3]
```diff
@@ -45,7 +50,13 @@ app.post('/upload', function(req, res) {
         coverFile = coverFile[coverFile.length - 1];
       }
       console.log(coverFile);
-      coverFile.mv(path.join(__dirname, '_submissions/cover/' + shortname + '.png'), err => console.log(err));
+      coverFile.mv(path.join(__dirname, '_submissions/cover/' + shortname + '.png'), err => console.error(err));
+      // Transcoding audio file
+      try {
+        ffmpeg(audioPath).save(transcodedAutioPath);
```
Comment (tidoust) [inline index.js]:
```suggestion
        ffmpeg(audioPath).save(transcodedAudioPath);
```

File: public/recorder.js (hunk 1 above)
Comment (tidoust) [reply public/recorder.js:122]: ¯\\\_(ツ)_/¯

**Review by tidoust — APPROVED** [review]

**Pull request merged**